### PR TITLE
feat(lib): Adds abac based KAO configuration

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
- Adds a new parameter to the encrypt method, `attributeValues`, which takes the output of the `GetAttributeValuesByFqns` gRPC call as its input.
- This will be followed by a more complete solution which also performs the RPC call if required: https://github.com/opentdf/client-web/pull/351

### CODE REVIEW

File: lib/src/policy/granter.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+type BooleanOperator = BooleanClause['op'];
+
+type ComplexBooleanClause = {
+  op: BooleanOperator;
+  children: BooleanClause[];
+};
+
+export function plan(dataAttrs: Value[]): KeyAccessTemplate[] {
+  // kases by value
```
Comment (pflynn-virtru) [inline lib/src/policy/granter.ts:38]: nit upper case KAS

**Review by pflynn-virtru — DISMISSED** [review]
code looks fine.  needs a split key expert to review

File: lib/src/policy/granter.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
+import { Attribute, AttributeRuleType, KeyAccessServer, Value } from './attributes.js';
+
+export type KeyAccessTemplate = {
```
Comment (elizabethhealy) [inline lib/src/policy/granter.ts:3]: nit: maybe rename to keysplitstep to keep in line with other sdks https://github.com/opentdf/platform/blob/d9a07f84ab5686fe13af82435af3201042dd7228/sdk/granter.go#L31

**Review by elizabethhealy — DISMISSED** [review]

File: lib/src/policy/granter.ts (hunk 2 above)
Comment (dmihalcik-virtru) [reply lib/src/policy/granter.ts]: Done

**Review by elizabethhealy — APPROVED** [review]

**Pull request merged**